File: src/client/src/services/model/currencyPairUpdate.js
import UpdateType from './updateType';
import CurrencyPair from './currencyPair';

export default class CurrencyPairUpdate {
  _updateType:UpdateType;
  _currencyPair:CurrencyPair;

  constructor(updateType:UpdateType, currencyPair:CurrencyPair) {
    this._updateType = updateType;
    this._currencyPair = currencyPair;
  }

  get updateType():UpdateType {
    return this._updateType;
  }

  get currencyPair():CurrencyPair {
    return this._currencyPair;
  }
}

File: src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Models/ReferenceData/CurrencyPairUpdate.cs
namespace Adaptive.ReactiveTrader.Client.Domain.Models.ReferenceData
{
    internal class CurrencyPairUpdate : ICurrencyPairUpdate
    {
        public CurrencyPairUpdate(UpdateType updateType, ICurrencyPair currencyPair)
        {
            UpdateType = updateType;
            CurrencyPair = currencyPair;
        }

        public UpdateType UpdateType { get; }
        public ICurrencyPair CurrencyPair { get; }
    }
}